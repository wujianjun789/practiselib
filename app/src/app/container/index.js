/**
 * Created by a on 2017/6/29.
 */
import React, { Component } from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux'
import PropTypes from 'prop-types';
import '../../../public/styles/app.less';
import Card from './Card';
import UserCenter from '../../common/containers/UserCenter';
import Overlayer from '../../common/containers/Overlayer';
import {loginHandler} from '../../api/login'
import {getModule} from '../action/index'
/**
 * @param {String} title required
 * @param {String} name  required
 * @param {Array}  items required
 */
export class App extends Component{
    constructor(props){
        super(props);
        this.loginFail = this.loginFail.bind(this);
        this.loginSuccess = this.loginSuccess.bind(this);
    }

    componentWillMount(){
        console.log(localStorage)
        /*if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            
            this.props.router.push('/login')
            
        }else{

            this.props.actions.loginHandler(sessionStorage.username, sessionStorage.password, this.loginSuccess, this.loginFail);

        }*/
        // if(this.props.userCenter.islogin!=1){
        //     this.props.router.push('/login')
        // }
        const {actions} = this.props;
        actions && actions.getModule();
    }

    componentDidMount() {
    }

    componentWillReceiveProps(nextProps) {



    }

    loginFail(){
        this.setState({style:{visibility: 'visible'}})
    }

    loginSuccess(){
        this.props.router.push('/')
    }

    render(){
        const {title="StarRiver", name="智慧路灯管理系统", items} = this.props;
        return (
            <div className="app">
                <div className="header clearfix">
                    <div className="header-left clearfix">
                        <span className="icon icon-logo"></span>
                        <span className="tit">{title}</span>
                    </div>
                    <UserCenter router={this.props.router}/>
                    <div className="header-right clearfix">
                        <span className="name">{name}</span>
                    </div>
                </div>
                <div className="cont">
                    <ul className="clearfix">
                        {items.map((item, index)=><li key={item.key}><Card _key={item.key} title={item.title} link={item.link}/></li>)}
                    </ul>
                </div>
                <Overlayer />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    let app = state.app;
    return {
        title: app.title,
        name: app.name,
        items: app.items,
        userCenter:state.userCenter
    }
}

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        actions: bindActionCreators({
            getModule: getModule,
            loginHandler
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
    title: PropTypes.string,
    name: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            key: PropTypes.string.isRequired,
            title: PropTypes.string.isRequired,
            link: PropTypes.string.isRequired
        })
    ).isRequired
}