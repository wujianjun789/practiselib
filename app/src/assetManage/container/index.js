/**
 * Created by a on 2017/7/17.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import {injectIntl} from 'react-intl';

import '../../../public/styles/assetManage-model.less';
import '../../../public/styles/assetManage-statistics.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '../../data/assetModels';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';

import {sideBarToggled} from '../action/index';
import {treeViewNavigator} from '../../common/util/index';
class AssetManageIndex extends Component {
  constructor(props) {
    super(props);

    this.state = {
      pathname: ""
    }

    this.initModel = this.initModel.bind(this);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);

    this.getMathNode = this.getMathNode.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    this.initModel();
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !==  this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }
  componentDidUpdate(){
    // console.log(this.props);
    // const {location} = this.props;
    // if(location.pathname !== this.state.pathname){
    //   this.setState({pathname: location.pathname}, ()=>{
    //     this.initModel();
    //   });
    // }
  }
  initModel(){
    getModelData(() => {this.mounted && this.initTreeData();});
  }

  initTreeData() {
    TreeData.map(tree=>{
      return Object.assign({}, tree, {name:this.props.intl.formatMessage({id:tree.name})});
    })
    this.props.actions.treeViewInit(TreeData);
    treeViewNavigator(TreeData, this.props.router);

    const node = this.getMathNode(TreeData, this.props.location.pathname);
    if(node){
      this.onToggle(node);
    }
  }

  getMathNode(data, pathname){
    for(let i=0;i<data.length;i++){
      if(!data[i].children && data[i].link.indexOf(pathname) > -1){
        return data[i];
      }else{
        if(data[i].children){
          const node = this.getMathNode(data[i].children, pathname);
          if(node){
            return node;
          }
        }

      }
    }

    return null;
  }

  onToggle(node) {
    const {actions} = this.props;
    actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }

    return <div className={'container ' + 'asset-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName="app.asset.manage" router={this.props.router}/>
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById,
      sideBarToggled: sideBarToggled,
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(AssetManageIndex));