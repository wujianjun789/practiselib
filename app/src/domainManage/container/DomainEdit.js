/**
 * Created by a on 2017/4/19.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import '../../../public/styles/domainmanage.less';
import Content from '../../components/Content'
import SideBarInfo from '../../components/SideBarInfo'
import Table from '../../components/Table';
import SearchText from '../../components/SearchText'
import Page from '../../components/Page'

import DomainPopup from '../component/DomainPopup'
import ConfirmPopup from '../../components/ConfirmPopup'
import {overlayerShow, overlayerHide} from '../../common/actions/overlayer'

import Topology from './Topology';

import {getDomainList, getDomainCountByName, getDomainListByName, addDomain, updateDomainById, deleteDomainById} from '../../api/domain'

import Immutable from 'immutable';
import {getIndexByKey} from '../../util/index'

export class DomainEdit extends Component {
    constructor(props) {
        super(props);
        this.state = {
            listMode: true,
            collapse: false,
            selectDomain: {
                id:"domain",
                latlng:{lng: null, lat: null},
                position: [/*{
                    "device_id": 1,
                    "device_type": 'DEVICE',
                    lng: 121.49971691534425,
                    lat: 31.239658843127756
                }*/],
                parentId:null,
                data: [{
                    id: 1,
                    name: '上海市'
                }]
            },

            page: Immutable.fromJS({
                pageSize: 10,
                current: 1,
                total: 21
            }),

            search: Immutable.fromJS({placeholder: '输入域名称', value: ''}),
            data: Immutable.fromJS([{id:1,name: '上海市', parentId: null, parentName:'无'},
                {id:2, name: '闵行区', parentId:1, parentName: '上海市'},
                {id:3, name: '徐汇区', parentId:1, parentName: '上海市'}]),
            topologyRefresh:{parentId:null}
        }

        this.columns = [{id: 1, field: "name", title: "域名称"}, {id:2, field: "parentName", title: "上级域"}]
        this.domainList = [];

        this.onToggle = this.onToggle.bind(this);
        this.initTreeData = this.initTreeData.bind(this);
        this.collpseHandler = this.collpseHandler.bind(this);
        this.searchChange = this.searchChange.bind(this);
        this.tableClick = this.tableClick.bind(this);
        this.searchSubmit = this.searchSubmit.bind(this);
        this.topologyItemClick = this.topologyItemClick.bind(this);
        this.updateSelectDomain = this.updateSelectDomain.bind(this);

        this.pageChange = this.pageChange.bind(this);
        this.domainHandler = this.domainHandler.bind(this);

        this.destoryTopology = this.destoryTopology.bind(this);

        this.requestDomain = this.requestDomain.bind(this);
        this.requestSearch = this.requestSearch.bind(this);
        this.initPageSize = this.initPageSize.bind(this);
        this.initDomainList = this.initDomainList.bind(this);

        this.getDomainParentList = this.getDomainParentList.bind(this);
    }

    componentWillMount() {
        this.mounted = true;
        this.requestDomain();
        this.initTreeData();
        this.requestSearch();
    }

    componentWillReceiveProps(nextProps) {
        const {sidebarNode} = nextProps;
        if (sidebarNode) {
            this.onToggle(sidebarNode);
        }
    }

    componentWillUnmount() {
        this.mounted = false;
    }

    requestDomain(){
        getDomainList(data=>{if(this.mounted){this.domainList=data;this.domainList.unshift({id:null, name:"无"});}});
    }

    requestSearch(){
        const {search, page} = this.state
        let name = search.get('value');
        let cur = page.get('current');
        let size = page.get('pageSize');
        let offset = (cur-1)*size;
        getDomainCountByName(name, data=>{this.mounted && this.initPageSize(data)})
        getDomainListByName(name, offset, size, data=>{this.mounted && this.initDomainList(data)})
    }

    initPageSize(data){
        let page = this.state.page.set('total', data.count);
        this.setState({page:page});
    }

    initDomainList(data){
        let list = data.map(domain=>{
            return Object.assign({}, domain, {parentName:domain.parent?domain.parent.name:"无"})
        })


        this.setState({data:Immutable.fromJS(list)});

        if(data.length){
            this.updateSelectDomain(data[0])
        }
    }

    initTreeData() {

    }

    destoryTopology(){
        this.setState({topologyRefresh:{parentId:null}});
    }

    getDomainParentList(){
        const {selectDomain} = this.state;
        let domainList = this.domainList.slice(0);
        let domainId = selectDomain.data[0].id;
        if(domainId){
            let index = getIndexByKey(domainList, 'id', domainId);
            if(index>-1){
                domainList.splice(index, 1);
            }
        }

        return domainList;
    }

    domainHandler(id){
        const {listMode, selectDomain} = this.state
        const {actions} = this.props;
        switch(id){
            case 'add':
                actions.overlayerShow(<DomainPopup title={"添加域"} data={{domainId:"", domainName:"",
                lat:"", lng:"", prevDomain:''}}
                                                   domainList={{titleKey:'name', valueKey:'name', options:this.domainList}}
                                                   onConfirm={(data)=>{
                                                        let domain = {};
                                                        domain.name = data.domainName;
                                                        domain.geoType = 0;
                                                        domain.geoPoint = {lat:data.lat, lng:data.lng};
                                                        domain.parentId = data.prevDomain;

                                                        addDomain(domain, ()=>{actions.overlayerHide();this.requestDomain();listMode?this.requestSearch():this.setState({topologyRefresh:{parentId:data.prevDomain}});});
                                                   }} onCancel={()=>{actions.overlayerHide()}}/>);
                break;
            case 'update':
                let lat="", lng="",updateId="",name="";

                if(selectDomain.position && selectDomain.position.length){
                    let latlng = selectDomain.position[0];
                    lat = latlng.lat?latlng.lat:"";
                    lng = latlng.lng?latlng.lng:"";
                }

                if(selectDomain.data && selectDomain.data.length){
                    let data = selectDomain.data[0];
                    updateId = data.id;
                    name = data.name;
                }
                actions.overlayerShow(<DomainPopup title={"修改域属性"} data={{domainId:updateId, domainName:name,
                lat:lat, lng:lng, prevDomain:''}}
                                                              domainList={{titleKey:'name', valueKey:'name', options:this.getDomainParentList()}}
                                                              onConfirm={(data)=>{
                                                                    let domain = {};
                                                                    domain.id = data.domainId;
                                                                    domain.name = data.domainName;
                                                                    domain.geoType = 0;
                                                                    domain.geoPoint = {lat:data.lat, lng:data.lng};
                                                                    domain.parentId = data.prevDomain;
                                                                    updateDomainById(domain, ()=>{actions.overlayerHide();this.requestDomain();listMode?this.requestSearch():this.setState({topologyRefresh:{parentId:data.prevDomain}});});
                                                              }} onCancel={()=>{actions.overlayerHide()}}/>);
                break;
            case 'delete':
                let curId="";
                if(selectDomain.data && selectDomain.data.length){
                    let data = selectDomain.data[0];
                    curId = data.id;
                }
                actions.overlayerShow(<ConfirmPopup iconClass="icon_popup_delete" tips={"是否删除选中域？"}
                                                 cancel={()=>{actions.overlayerHide()}} confirm={()=>{deleteDomainById(curId,
                                                 ()=>{actions.overlayerHide();this.requestDomain();listMode?this.requestSearch():this.setState({topologyRefresh:{parentId:data.prevDomain}});})}}/>);
                break;
        }
    }

    pageChange(current, pageSize) {
        let page = this.state.page.set('current', current);
        this.setState({page: page}, ()=>{
            this.requestSearch();
        });
    }

    tableClick(row){
        this.updateSelectDomain(row.toJS());

    }

    topologyItemClick(data){
        this.updateSelectDomain(data);
    }

    updateSelectDomain(domain){
        let selectDomain = this.state.selectDomain;
        selectDomain.latlng = domain.geoPoint;
        selectDomain.position.splice(0)
        selectDomain.position.push(Object.assign({}, {"device_id":domain.id, "device_type":"DEVICE"}, domain.geoPoint))
        selectDomain.parentId = domain.parentId;
        selectDomain.data.splice(0);
        selectDomain.data.push({id:domain.id, name:domain.name});
        this.setState({selectDomain:selectDomain})
    }

    searchSubmit(){

        this.requestSearch();
    }

    searchChange(value){
        this.setState({search:this.state.search.update('value', v=>value)});
    }

    onToggle(node) {
        let mode = undefined;

        if(node.id == "list-mode"){
            mode = true;
        }else if(node.id == 'topology-mode'){
            mode = false;
        }

        mode != undefined && this.setState({listMode:mode});
    }

    collpseHandler() {
        this.setState({collapse: !this.state.collapse})
    }

    render() {
        const {listMode, collapse, selectDomain, page, search, data, topologyRefresh} = this.state
        return (
            <Content className={'offset-right '+(listMode?'list-mode ':'topology ')+(collapse?'collapsed':'')}>
                <div className="heading">
                    <SearchText placeholder={search.get('placeholder')} value={search.get('value')}
                                onChange={this.searchChange} submit={this.searchSubmit}/>
                    <button className="btn btn-default add-domain" onClick={()=>this.domainHandler('add')}>添加</button>
                </div>
                {
                    listMode ?
                        <div className="list-mode">
                            <div className="table-container">
                                <Table columns={this.columns} data={data} activeId={selectDomain.data[0].id}
                                       rowClick={this.tableClick}/>
                                <Page className="page" showSizeChanger pageSize={page.get('pageSize')}
                                      current={page.get('current')} total={page.get('total')} onChange={this.pageChange}/>
                            </div>
                        </div> :
                        <Topology topologyRefresh={topologyRefresh} callFun={this.destoryTopology} itemClick={this.topologyItemClick}/>
                }
                <SideBarInfo mapDevice={selectDomain} collpseHandler={this.collpseHandler}>
                    <div className="panel panel-default device-statics-info">
                        <div className="panel-heading">
                            <span className="icon_domain_property"></span>域属性
                        </div>
                        <div className="panel-body domain-property">
                            <span className="domain-name">{selectDomain.data[0].name}</span>
                            <button className="btn btn-primary pull-right" onClick={()=>this.domainHandler('update')}>编辑</button>
                            <button className="btn btn-danger pull-right" onClick={()=>this.domainHandler('delete')}>删除</button>

                        </div>
                    </div>
                </SideBarInfo>
            </Content>
        )
    }
}


function mapStateToProps(state) {
    return {
        sidebarNode: state.domainManage.get('sidebarNode')
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            overlayerShow: overlayerShow,
            overlayerHide: overlayerHide
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DomainEdit);