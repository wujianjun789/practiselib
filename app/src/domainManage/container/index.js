/**
 * Created by a on 2017/7/17.
 */
import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import HeadBar from '../../components/HeadBar'
import SideBar from '../../components/SideBar'
import Overlayer from '../../common/containers/Overlayer'

import {TreeData} from '.././../data/domainModel'
import {treeViewInit} from '../../common/actions/treeView'
import {sideBarToggled} from '../action/index'
class DomainManageIndex extends Component{
    constructor(props){
        super(props);
        this.initTreeData = this.initTreeData.bind(this);
        this.onToggle = this.onToggle.bind(this);
    }

    componentWillMount(){
        if(sessionStorage.sessionID==0||sessionStorage.sessionID==null||sessionStorage.sessionID==""){
            this.props.router.push('/login') 
        }else{}
        this.initTreeData();
    }

    componentWillUnmount(){

    }

    componentDidMount(){
        console.log(this.props.userCenter); 
    }

    initTreeData(){
        this.props.actions.treeViewInit(TreeData);
    }

    onToggle(node){
        this.props.actions.sideBarToggled(node);
    }

    render(){
        let path = "";
        const {children} = this.props;
        if(children){
            path = children.props.route.path;
        }
        return <div className={"container "+"domain-"+path}>
            <HeadBar moduleName={/*path=="domainEdit"?*/"域管理"/*:"地图预览"*/} router={this.props.router}/>
            <SideBar onToggle={this.onToggle}/>
            {this.props.children}
            <Overlayer />
        </div>
    }
}

function mapStateToProps(state) {
    return {
        userCenter:state.userCenter
    }
}

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            treeViewInit: treeViewInit,
            sideBarToggled: sideBarToggled
        }, dispatch)
    }
}
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(DomainManageIndex);