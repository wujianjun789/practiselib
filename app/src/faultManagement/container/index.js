import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import Table from '../../components/Table';
import Page from '../../components/Page';
import SideBarInfo from '../../components/SideBarInfo';
import Select from '../../components/Select.1';
import ConfirmPopup from '../../components/ConfirmPopup';
import Immutable from 'immutable';
import { overlayerShow, overlayerHide } from '../../common/actions/overlayer';

import Content from '../../components/Content';
import { treeViewInit } from '../../common/actions/treeView';
import { injectIntl } from 'react-intl';
import { getChildDomainList } from '../../api/domain';
import { DatePicker } from 'antd';
import moment from 'moment';
import PieChart from '../../lightManage/utils/pieChart';
import { spliceInArray } from '../../util/algorithm';
import { getFaultOrAlertList } from '../../api/asset';

export class Fault extends Component {
  constructor(props) {
    super(props);
    this.state = {
      model: '',
      collapse: false,
      infoCollapse: false,
      page: Immutable.fromJS({
        pageSize: 15,
        current: 1,
        total: 0
      }),
      data: Immutable.fromJS([]),
      domainList: {
        titleField: 'name',
        valueField: 'name',
        index: 0,
        value: '',
        options: []
      },
      levelList: {
        titleField: 'title',
        valueField: 'name',
        index: 0,
        value: 'deadly',
        options: [
          {
            title: '致命',
            name: 'deadly'
          },
          {
            title: '严重',
            name: 'seriousness'
          },
          {
            title: '一般',
            name: 'ordinary'
          },
          {
            title: '提示',
            name: 'energy'
          }
        ]
      },
      start: moment(),
      end: moment(),
      statisticalInfo: {
        deadly: 50,
        serious: 30,
        general: 10,
        hint: 10
      }
    };

    this.columns = [
      {
        id: 0,
        field: 'model',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.device.model'
        })
      },
      {
        id: 1,
        field: 'name',
        title: this.props.intl.formatMessage({ id: 'app.device.name' })
      },
      {
        id: 3,
        field: 'param',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.param' })
      },
      {
        id: 3,
        field: 'level',
        title: this.props.intl.formatMessage({ id: 'sysOperation.alarm.level' })
      },
      {
        id: 4,
        field: 'threshold',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.threshold'
        })
      },
      {
        id: 5,
        field: 'value',
        title: this.props.intl.formatMessage({
          id: 'sysOperation.alarm.test.value'
        })
      },
      {
        id: 6,
        field: 'time',
        title: this.props.intl.formatMessage({ id: 'sysOperation.fault.time' })
      }
    ];
  }

  componentWillMount() {
    this.mounted = true;
    this.setState({ model: this.props.params.fault }, this.requestSearch);
    getChildDomainList(data => {
      this.mounted && this.initDomainList(data);
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidUpdate() {
    const { sidebarNode } = this.props;
    if (
      sidebarNode &&
      !sidebarNode.children &&
      sidebarNode.id != this.state.model
    ) {
      this.setState({ model: sidebarNode.id }, this.requestSearch);
    }
  }

  formatIntl = formatId => {
    const { intl } = this.props;
    return intl ? intl.formatMessage({ id: formatId }) : null;
  };

  initDomainList = data => {
    let domainList = Object.assign(
      {},
      this.state.domainList,
      { index: 0 },
      { value: data.length ? data[0].name : '' },
      { options: data }
    );
    this.setState({ domainList: domainList }, this.requestSearch);
  };

  requestSearch = () => {
    let model = '';
    switch (this.state.model) {
      case 'alarm':
        model = 'alerts';
        break;
      case 'fault':
        model = 'faults';
        break;
    }
    const { domainList, levelList, start, end, page } = this.state;
    const { pageSize, current } = page.toJS();
    if (domainList.options.length) {
      const domainId = domainList.options[domainList.index]['id'];
      // const level = levelList.options[levelList.index]['name'];
      const level = levelList.index;
      const offset = pageSize * (current - 1);
      const limit = pageSize;
      getFaultOrAlertList(
        model,
        domainId,
        level,
        start,
        end,
        offset,
        limit,
        res => {
          this.setState({ data: res });
        }
      );
    }
  };

  initPageSize = data => {
    let page = this.state.page.set('total', data.count);
    this.setState({
      page: page
    });
  };

  pageChange = (current, pageSize) => {
    let page = this.state.page.set('current', current);
    this.setState(
      {
        page: page
      },
      () => {
        this.requestSearch();
      }
    );
  };

  collapseHandler = id => {
    this.setState({
      [id]: !this.state[id]
    });
  };

  selectChange = (event, key) => {
    let index = event.target.selectedIndex;
    let list = this.state[key];
    list.index = index;
    list.value = list.options[index].name;
    this.setState({ [key]: list }, () => {
      this.requestSearch();
    });
  };

  dateChange = (id, value) => {
    this.setState({ [id]: value }, () => {
      const { start, end } = this.state;
      let prompt = start && end && end.isBefore(start);
      this.setState({
        prompt: Object.assign({}, this.state.prompt, { date: prompt })
      });
    });
  };

  drawChart = ref => {
    if (ref == null) {
      this.piechart.destroy();
    } else {
      const { statisticalInfo } = this.state;
      let data = [
        statisticalInfo.deadly,
        statisticalInfo.serious,
        statisticalInfo.general,
        statisticalInfo.hint
      ];
      this.piechart = new PieChart({
        wrapper: ref,
        data: data,
        color: ['#F83D59', '#FA919C', '#FA9E17', '#FBEF35']
      });
    }
  };

  render() {
    const {
      collapse,
      infoCollapse,
      page,
      data,
      domainList,
      typeList,
      levelList,
      start,
      end,
      statisticalInfo,
      allChecked,
      checked
    } = this.state;
    return (
      <Content className={'offset-right ' + (collapse ? 'collapsed' : '')}>
        <div className="heading">
          <Select
            id="domain"
            titleField={domainList.titleField}
            valueField={domainList.valueField}
            options={domainList.options}
            value={domainList.value}
            onChange={e => {
              this.selectChange(e, 'domainList');
            }}
          />
          {/* <Select id="type" titleField={typeList.titleField} valueField={typeList.valueField}
          options={typeList.options} value={typeList.value} onChange={(e) => {this.selectChange(e, 'typeList');}}/> */}
          <Select
            id="level"
            titleField={levelList.titleField}
            valueField={levelList.valueField}
            options={levelList.options}
            value={levelList.value}
            onChange={e => {
              this.selectChange(e, 'levelList');
            }}
          />
          <div className="datePicker">
            <DatePicker
              id="startDate"
              format="YYYY/MM/DD"
              placeholder="开始日期"
              style={{ width: '106px' }}
              defaultValue={start}
              value={start}
              onChange={value => this.dateChange('start', value)}
            />
            <span>-</span>
            <DatePicker
              id="endDate"
              format="YYYY/MM/DD"
              placeholder="结束日期"
              style={{ width: '106px' }}
              defaultValue={end}
              value={end}
              onChange={value => this.dateChange('end', value)}
            />
          </div>
        </div>
        <div className="table-container">
          <Table columns={this.columns} data={data} />
          <Page
            className={'page ' + (page.get('total') == 0 ? 'hidden' : '')}
            pageSize={page.get('pageSize')}
            current={page.get('current')}
            total={page.get('total')}
            onChange={this.pageChange}
          />
        </div>
        <SideBarInfo
          collapseHandler={this.collapseHandler}
          className={infoCollapse ? 'infoCollapse ' : ''}
        >
          <div className="panel panel-default statistical-info">
            <div
              className="panel-heading"
              role="presentation"
              onClick={() => {
                !collapse && this.collapseHandler('infoCollapse');
              }}
            >
              <span className="icon_chart" />
              {this.props.intl.formatMessage({
                id: 'sysOperation.statistical.info'
              })}
              <span className="icon icon_collapse pull-right" />
            </div>
            <div className={'panel-body' + (infoCollapse ? 'collapsed' : '')}>
              <div className="left">
                <div id="alarm" className="circle" ref={this.drawChart} />
              </div>
              <div className="right">
                <div className="count deadly">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.deadly')} :{' '}
                  {statisticalInfo.deadly
                    ? statisticalInfo.deadly
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count serious">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.serious')} :{' '}
                  {statisticalInfo.serious
                    ? statisticalInfo.serious
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count general">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.general')} :{' '}
                  {statisticalInfo.general
                    ? statisticalInfo.general
                    : this.formatIntl('sysOperation.noCount')}
                </div>
                <div className="count hint">
                  <div className="dot" />
                  {this.formatIntl('sysOperation.count.hint')} :{' '}
                  {statisticalInfo.hint
                    ? statisticalInfo.hint
                    : this.formatIntl('sysOperation.noCount')}
                </div>
              </div>
            </div>
          </div>
        </SideBarInfo>
      </Content>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    sidebarNode: state.systemOperation.get('sidebarNode')
  };
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      treeViewInit,
      overlayerShow,
      overlayerHide
    },
    dispatch
  )
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(Fault));
