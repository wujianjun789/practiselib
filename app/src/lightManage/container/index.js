/**
 * Created by a on 2017/8/24.
 */
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import '../../../public/styles/lightManage-map.less';
import '../../../public/styles/lightManage-control.less';

import HeadBar from '../../components/HeadBar';
import SideBar from '../../components/SideBar';
import Overlayer from '../../common/containers/Overlayer';

import {getModelData, TreeData} from '.././../data/lightModel';
import {treeViewInit, onToggleById} from '../../common/actions/treeView';
// import {sideBarToggled} from '../action/index'
class lightManageIndex extends Component {
  constructor(props) {
    super(props);
    this.initTreeData = this.initTreeData.bind(this);
    this.onToggle = this.onToggle.bind(this);
  }

  componentWillMount() {
    this.mounted = true;
    getModelData(null, () => {
      this.mounted && this.initTreeData();
    });
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  componentDidMount() {
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.location.pathname !== this.props.location.pathname){
      this.props.actions.onToggleById(nextProps.location.pathname);
    }
  }

  initTreeData() {
    this.props.actions.treeViewInit(TreeData);
  }

  onToggle(node) {
    // console.log(node);
    // this.props.actions.sideBarToggled(node);
  }

  render() {
    let parentPath = '';
    let childPath = '';
    const {routes} = this.props;
    if (routes.length > 4) {
      parentPath = routes[4].path;
    }

    if (routes.length > 5) {
      childPath = routes[5].path;
    }
    return <div className={'container light-' + parentPath + ' ' + parentPath + '-' + childPath}>
      <HeadBar moduleName={'app.light'} router={this.props.router}/>
                    
      <SideBar onToggle={this.onToggle}/>
      {this.props.children}
      <Overlayer />
    </div>;
  }
}

function mapStateToProps(state) {
  return {
    userCenter:state.userCenter,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      treeViewInit: treeViewInit,
      onToggleById: onToggleById
      // sideBarToggled: sideBarToggled
    }, dispatch),
  };
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(lightManageIndex);