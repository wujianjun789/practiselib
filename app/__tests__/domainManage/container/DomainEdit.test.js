/**
 * Created by a on 2017/7/27.
 */
import React from 'react';
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import '../../../public/js/jquery-3.1.1.min'
import '../../../public/leaflet/leaflet';
window.d3 = require('../../../public/js/d3.min')

import {DomainEdit} from '../../../src/domainManage/container/DomainEdit';


test('DomainEdit renders', ()=>{
    const component = renderer.create(
        <DomainEdit />
    )

    let domainEdit = component.toJSON();
    expect(domainEdit).toMatchSnapshot();
})

test('DomainEdit div click', ()=>{
    const component = shallow(<DomainEdit />)
    expect(component.find('.heading').length).toEqual(1);
    expect(component.find('.table-container').length).toEqual(1);
    
    expect
})