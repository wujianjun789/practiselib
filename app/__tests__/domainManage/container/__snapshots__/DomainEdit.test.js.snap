exports[`test DomainEdit renders 1`] = `
<div
  className="content offset-right list-mode ">
  <div
    className="heading">
    <div
      className="searchText "
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}>
      <input
        onChange={[Function]}
        placeholder="输入域名称"
        type="search"
        value="" />
      <ul
        className="" />
      <span
        className="glyphicon glyphicon-search"
        onClick={[Function]} />
    </div>
    <button
      className="btn btn-default add-domain"
      onClick={[Function]}>
      添加
    </button>
  </div>
  <div
    className="list-mode">
    <div
      className="table-container">
      <div
        className="table-responsive ">
        <table
          className="table table-hover">
          <thead>
            <tr>
              <th
                className="hidden">
                <input
                  checked={undefined}
                  onChange={[Function]}
                  type="checkbox" />
              </th>
              <th>
                域名称
              </th>
              <th>
                上级域
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
      <ul
        className="rc-pagination page hidden"
        style={Object {}}
        unselectable="unselectable">
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-prev"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="上一页">
          <a>
            
          </a>
        </li>
        <li
          aria-disabled={true}
          className="rc-pagination-disabled rc-pagination-next"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title="下一页">
          <a>
            
          </a>
        </li>
        <div
          className="rc-pagination-options" />
      </ul>
    </div>
  </div>
  <div
    className="container-fluid sidebar-info ">
    <div
      className="row collapse-container"
      onClick={[Function]}>
      <span
        className="icon_verital" />
    </div>
    <div
      className="panel panel-default device-statics-info">
      <div
        className="panel-heading">
        <span
          className="icon_domain_property" />
        域属性
      </div>
      <div
        className="panel-body domain-property">
        <span
          className="domain-name">
          
        </span>
        <button
          className="btn btn-primary pull-right"
          disabled={true}
          onClick={[Function]}>
          编辑
        </button>
        <button
          className="btn btn-danger pull-right"
          disabled={true}
          onClick={[Function]}>
          删除
        </button>
      </div>
    </div>
    <div
      className="panel panel-default map-position">
      <div
        className="panel-heading">
        <span
          className="icon_map_position" />
        地图位置
      </div>
      <div
        className="map-container panel-body">
        <div
          className="map-view "
          id="domain" />
      </div>
    </div>
  </div>
</div>
`;
